src/libPMacc/include/preprocessor/struct.hpp

```diff
@@ -0,0 +1,289 @@
+    )
+
+/** generate the definition of a struct
+ *
+ * @param namespace_name name of a unique namespace to avoid naming conflicts
+ * @param name name of the struct
+ * @param ... preprocessor sequence with TypeMemberPair's
+ */
+#define PMACC_PP_STRUCT_DEF(namespace_name,name,...)                           \
+namespace namespace_name{                                                      \
+    BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_VECTOR_DEF,__VA_ARGS__) \
+    struct EmptyStruct{};                                                      \
+    struct EmptyStruct2{};                                                     \
+    struct name : private EmptyStruct, private EmptyStruct2 {                  \
+        name():                                                                \
+        PMACC_PP_INIT_VALUE_VARIABLES(PMACC_PP_X_INIT_VALUE_VARIABLE,__VA_ARGS__),                            \
+        PMACC_PP_INIT_VALUE_VARIABLES(PMACC_PP_X_INIT_VALUE_VARIABLE_WITH_PAREN,__VA_ARGS__){}                \
+                                                                                                              \
+        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_VALUE_VARIABLE,__VA_ARGS__)  \
+        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_VALUE_VARIABLE,__VA_ARGS__)    \
+        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_VECTOR_VARIABLE,__VA_ARGS__) \
+        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_STRING_VARIABLE,__VA_ARGS__) \
+	    BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_EXTENT,__VA_ARGS__)          \
+        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_VALUE_VARIABLE_WITH_PAREN,__VA_ARGS__)    \
+        };                                                                     \
+    BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_INIT_C_STRING_VARIABLE,__VA_ARGS__);      \
+}  /*namespace*/                                                               \
+using namespace_name::name
+
+
+/** add data to TypeMemberPair's
+ *
+ * transform (typeId,(value_type,name,initValue,...) to (typeId,(data,value_type,name,initValue,...)
+ *
+ * @param data any data which should be added to the TypeMemberPair's
+ * @param ... preprocessor sequence with TypeMemberPair's
+ */
+#define PMACC_PP_ADD_DATA_TO_TYPEDESCRIPTION(data,...) BOOST_PP_SEQ_FOR_EACH(PMACC_PP_ADD_DATA_TO_TYPEDESCRIPTION_MACRO,data,__VA_ARGS__)
+
+/** generate a struct with static and dynamic members
+ *
+ * @param name name of the struct
+ * @param ... preprocessor sequence with TypeMemberPair's e.g. (PMACC_C_VALUE(int,a,2))
+ */
+#define PMACC_STRUCT(name,...)                                                 \
+    PMACC_PP_STRUCT_DEF(BOOST_PP_CAT(BOOST_PP_CAT(pmacc_,name),__COUNTER__),name,PMACC_PP_ADD_DATA_TO_TYPEDESCRIPTION(name,__VA_ARGS__))
```

Comment:
no problem, thx.
git and also the github gui would show you a small red sign in the line if it's missing.